### PR TITLE
Uberv2 optimization

### PR DESCRIPTION
**Pull request by AlexanderVeselov**
- Builds on Mac
- Fix some warnings

### CODE REVIEW

File: BaikalIO/material_io.cpp [hunk 1]
```diff
@@ -227,13 +227,13 @@ namespace Baikal
...
-        material->LinkRefractionIOR(element.Attribute("refraction_link_ior") == "true");
-        material->SetMultiscatter(element.Attribute("sss_multyscatter") == "true");
+        material->SetDoubleSided(strcmp(element.Attribute("emission_doublesided"), "true") == 0);
+        material->LinkRefractionIOR(strcmp(element.Attribute("refraction_link_ior"), "true") == 0);
+        material->SetMultiscatter(strcmp(element.Attribute("sss_multyscatter"), "true") == 0);
         material->SetName(name);
 
         auto num_inputs = material->GetNumInputs();
-        for (size_t a = 0; a < num_inputs; ++a)
+        for (uint32_t a = 0u; a < num_inputs; ++a)
```
Comment (Kvalme) [inline BaikalIO/material_io.cpp:236]: What is the reason for this change?
num_inputs is size_t since GetNumInputs() return size_z

File: BaikalStandalone/Application/application.cpp [hunk 2]
```diff
@@ -953,7 +953,7 @@ namespace Baikal
 
                 if (!m_object_name.empty())
                 {
-                    ImGui::Text(m_object_name.c_str());
+                    ImGui::Text("%s", m_object_name.c_str());
```
Comment (Kvalme) [inline BaikalStandalone/Application/application.cpp:956]: There was warning for this?

File: Rpr/WrapObject/ContextObject.cpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ ContextObject::ContextObject(rpr_creation_flags creation_flags)
 {
     rpr_int result = RPR_SUCCESS;
 
-    bool interop = creation_flags & RPR_CREATION_FLAGS_ENABLE_GL_INTEROP;
+    bool interop = static_cast<bool>(creation_flags & RPR_CREATION_FLAGS_ENABLE_GL_INTEROP);
```
Comment (Kvalme) [inline Rpr/WrapObject/ContextObject.cpp:112]: it would be better not to cast but to make check:
bool interop = ((creation_flags & RPR_CREATEION_FLAGS_ENABLE_GL_INTEROP) != 0)

**Review by Kvalme — CHANGES_REQUESTED** [review]

File: BaikalIO/material_io.cpp (hunk 1 above)
Comment (AlexanderVeselov) [reply BaikalIO/material_io.cpp:236]: Material::GetInput requires std::uint32_t. Maybe it would be better to change its parameter type to size_t? It works with std::vector and vector::value_type is size_t.

File: BaikalStandalone/Application/application.cpp (hunk 2 above)
Comment (AlexanderVeselov) [reply BaikalStandalone/Application/application.cpp:956]: Yes, Mac's compiler gave me a warning about insecurity of calling this without "%s".
